verilog/tools/ls/symbol-table-handler.cc

```diff
@@ -248,7 +251,43 @@ std::vector<verible::lsp::Location> SymbolTableHandler::FindDefinitionLocation(
   const verible::TextStructureView &text = parsedbuffer->parser().Data();
 
   const verible::TokenInfo cursor_token = text.FindTokenAt(cursor);
-  auto symbol = cursor_token.text();
+  return cursor_token.text();
+}
+
+std::optional<verible::lsp::Location>
+SymbolTableHandler::GetLocationFromSymbolTableNode(
+    absl::string_view symbol, const VerilogSourceFile *file_origin) {
+  // TODO (glatosinski) add iterating over multiple definitions
+  verible::lsp::Location location;
+  if (!file_origin && curr_project_) {
+    file_origin = curr_project_->LookupFileOrigin(symbol);
+  }
+  if (!file_origin) {
+    LOG(ERROR) << "Origin file not available";
+    return {};
+  }
+  location.uri = PathToLSPUri(file_origin->ResolvedPath());
+  auto *textstructure = file_origin->GetTextStructure();
+  if (!textstructure) {
+    LOG(ERROR) << "Origin file's text structure is not parsed";
+    return {};
+  }
+  verible::LineColumnRange symbollocation =
+      textstructure->GetRangeForText(symbol);
+  location.range.start = {.line = symbollocation.start.line,
+                          .character = symbollocation.start.column};
+  location.range.end = {.line = symbollocation.end.line,
+                        .character = symbollocation.end.column};
+  return location;
+}
+
+std::vector<verible::lsp::Location> SymbolTableHandler::FindDefinitionLocation(
```

Comment:
why does this function returns an array of locations when all it ever does is return a single location ?
I suspect it should in the future return possibly multiple locations, but the corresponding comment got lost.
 `// TODO add iterating over multiple definitions?`

Reply:
I brought back the TODO comment. Originally the `std::vector<verible::lsp::Location>` was supposed to reflect how the `textDocument/definition` output looks like (a list of found definitions). In the original implementation it returns at most one definition. In the https://github.com/chipsalliance/verible/pull/1678, this method will return all places contributing to symbol definition (for multiline definitions).

Reply:
I'd leave it as a vector for now, since it should be updated soon.